### PR TITLE
Update to Gu.Roslyn.Assert 4.2.0

### LINKED ISSUE #446: Move to latest Gu.Roslyn.Assert nuget packages
This is preparation work for #416 

The equivalent code for `DiagnosticsSuppressorAnalyzer` was now moved to [Gu.Roslyn.Assert](https://github.com/GuOrg/Gu.Roslyn.Asserts/pull/255)
Removed the original class from our source and call `RoslynAssert.Suppressed` instead.

The latest Gu.Roslyn.Assert has nullable enabled by default. This requires some changes to other tests.

### PR DESCRIPTION
**Pull request by manfred-brands**
Fixes #446 

Still in draft, waiting for the following PRs to be merged and the new version of these packages to be released.
* https://github.com/GuOrg/Gu.Roslyn.Asserts/pull/261
* https://github.com/GuOrg/Gu.Roslyn.Asserts/pull/259

### CODE REVIEW

**Comment by manfred-brands** [comment]
@mikkelbu This is yet another step closer to #416.
It uses the latest version of Gu.Roslyn.Assert. This has nullability enabled by default.
I also transferred my own testing class for Suppressors to a Gu.Roslyn.Assert matching API and moved it to that repository.
This allows re-using it in other Suppression Analyzers.

Although my Gu.Roslyn.Assert.Analyzers changes are merged into master, they were forgotten to be released.
Give John a day or two, then I remove the last commit.

File: src/nunit.analyzers.tests/DiagnosticSuppressors/DereferencePossiblyNullReferenceSuppressorTests.cs [hunk 1]
```diff
@@ -240,50 +234,50 @@ public void Test([Values] bool create)
                 {ABDefinition}
```
Comment (mikkelbu) [inline src/nunit.analyzers.tests/DiagnosticSuppressors/DereferencePossiblyNullReferenceSuppressorTests.cs:234]: `↓` in line 231 - `Assert.That(↓a.B.Text.Length, Is.GreaterThan(0));`

File: src/nunit.analyzers.tests/DiagnosticSuppressors/DereferencePossiblyNullReferenceSuppressorTests.cs [hunk 2]
```diff
@@ -638,9 +612,9 @@ public void HasCountShouldNotAffectNullabilitySuppression(List<int>? maybeNull)
                 }
```
Comment (mikkelbu) [inline src/nunit.analyzers.tests/DiagnosticSuppressors/DereferencePossiblyNullReferenceSuppressorTests.cs:612]: Line 610 we should add `↓` in `Assert.That(↓maybeNull[1], Is.EqualTo(1));`.

File: src/nunit.analyzers.tests/DiagnosticSuppressors/DereferencePossiblyNullReferenceSuppressorTests.cs [hunk 3]
```diff
@@ -638,9 +612,9 @@ public void HasCountShouldNotAffectNullabilitySuppression(List<int>? maybeNull)
...
             ", @"using System.Collections.Generic;");
 
-            await DiagnosticsSuppressorAnalyzer.EnsureSuppressed(suppressor,
-                DereferencePossiblyNullReferenceSuppressor.SuppressionDescriptors["CS8604"], testCode)
-                .ConfigureAwait(false);
+            RoslynAssert.Suppressed(suppressor,
+                ExpectedDiagnostic.Create(DereferencePossiblyNullReferenceSuppressor.SuppressionDescriptors["CS8602"]),
```
Comment (mikkelbu) [inline src/nunit.analyzers.tests/DiagnosticSuppressors/DereferencePossiblyNullReferenceSuppressorTests.cs:616]: Just curious. In moving the testing class to Gu.Roslyn.Assert was it improved so this noticed the wrong descriptor or did something else change ?

**Review by mikkelbu — APPROVED** [review]
Great PR @manfred-brands and good that your testing class can be used more widely via Gu.Roslyn.Assert 👍. I've mostly added some comments about `↓` that we can forgotten earlier.

File: src/nunit.analyzers.tests/DiagnosticSuppressors/DereferencePossiblyNullReferenceSuppressorTests.cs [hunk 4]
```diff
@@ -91,16 +90,15 @@ public void Test()
                 public void SetS(string? v) => s = v;
```
[the remark this answers is not in the document]
Comment (manfred-brands) [reply src/nunit.analyzers.tests/DiagnosticSuppressors/DereferencePossiblyNullReferenceSuppressorTests.cs:90]: The overload that accepts an `ExpectedDiagnostic` does not warn for this as the `ExpectedDiagnostic` could contain a location. Maybe the code should check if there is no location specified, but that would be a breaking change to that library.

File: src/nunit.analyzers.tests/DiagnosticSuppressors/DereferencePossiblyNullReferenceSuppressorTests.cs [hunk 5]
```diff
@@ -137,16 +134,15 @@ private static void DoSomething(string s)
                 }
```
[the remark this answers is not in the document]
Comment (manfred-brands) [reply src/nunit.analyzers.tests/DiagnosticSuppressors/DereferencePossiblyNullReferenceSuppressorTests.cs:134]: Done.

File: src/nunit.analyzers.tests/DiagnosticSuppressors/DereferencePossiblyNullReferenceSuppressorTests.cs [hunk 6]
```diff
@@ -159,53 +155,52 @@ public void Test()
                 private static int? GetNext() => 1;
```
[the remark this answers is not in the document]
Comment (manfred-brands) [reply src/nunit.analyzers.tests/DiagnosticSuppressors/DereferencePossiblyNullReferenceSuppressorTests.cs:155]: Done.

File: src/nunit.analyzers.tests/DiagnosticSuppressors/DereferencePossiblyNullReferenceSuppressorTests.cs (hunk 1 above)
Comment (manfred-brands) [reply src/nunit.analyzers.tests/DiagnosticSuppressors/DereferencePossiblyNullReferenceSuppressorTests.cs:234]: Done.

File: src/nunit.analyzers.tests/DiagnosticSuppressors/DereferencePossiblyNullReferenceSuppressorTests.cs [hunk 7]
```diff
@@ -219,16 +214,15 @@ public void Test([Values] bool create)
                 {ABDefinition}
```
[the remark this answers is not in the document]
Comment (manfred-brands) [reply src/nunit.analyzers.tests/DiagnosticSuppressors/DereferencePossiblyNullReferenceSuppressorTests.cs:214]: Done.

File: src/nunit.analyzers.tests/DiagnosticSuppressors/DereferencePossiblyNullReferenceSuppressorTests.cs (hunk 2 above)
Comment (manfred-brands) [reply src/nunit.analyzers.tests/DiagnosticSuppressors/DereferencePossiblyNullReferenceSuppressorTests.cs:612]: Done.

File: src/nunit.analyzers.tests/DiagnosticSuppressors/DereferencePossiblyNullReferenceSuppressorTests.cs (hunk 3 above)
Comment (manfred-brands) [reply src/nunit.analyzers.tests/DiagnosticSuppressors/DereferencePossiblyNullReferenceSuppressorTests.cs:616]: Yes, it previously checked that the errors were suppressible, not if it was matching the id.
In the final test it only checked that the error was suppressed by this suppressor with the Justification.
The Justification (Assert.NotNull, etc) is the same regardless of CS error.

File: src/nunit.analyzers.tests/DiagnosticSuppressors/DereferencePossiblyNullReferenceSuppressorTests.cs [hunk 8]
```diff
@@ -114,16 +112,15 @@ private static string DoSomething(string? s)
                 }
```
[the remark this answers is not in the document]
Comment (manfred-brands) [reply src/nunit.analyzers.tests/DiagnosticSuppressors/DereferencePossiblyNullReferenceSuppressorTests.cs:112]: Done.

**Comment by manfred-brands** [comment]
Thanks again @mikkelbu for your review.
Changes made.
I found a few more places to insert a ↓ character.

**Review by mikkelbu — APPROVED** [review]
Great work @manfred-brands. I'm happy to merge it now if you are.

**Comment by manfred-brands** [comment]
Thanks @mikkelbu. Ready for merge.

**Pull request merged**